### PR TITLE
Adds allergies exercise

### PR DESCRIPTION
**Pull request by amscotti**
Hi, this PR is to add a new exercise for `allergies`, [here is a link to the problem-specifications](https://github.com/exercism/problem-specifications/tree/master/exercises/allergies)

I'm still learning Dart, so feel free to point out anything that needs to be change or improved. I'm also happy to change the `config.json` to better fit your layout for the track.

Thanks! :)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -381,6 +381,19 @@
         "sorting",
         "trees"
       ]
+    },
+    {
+      "slug": "allergies",
+      "uuid": "729f4c8b-678a-434c-a09e-a71aca4bb2c0",
+      "core": false,
+      "unlocked_by": null,
```
[the remark this answers is not in the document]
Comment (Stargator) [reply config.json]: @amscotti This can be reverted and it can be unlocked by difference of squares.

[the remark this answers is not in the document]
Comment (amscotti) [reply config.json]: @Stargator Done

File: config.json [hunk 2]
```diff
@@ -381,6 +381,19 @@
         "sorting",
         "trees"
       ]
+    },
+    {
+      "slug": "allergies",
+      "uuid": "729f4c8b-678a-434c-a09e-a71aca4bb2c0",
+      "core": false,
+      "unlocked_by": "binary-search-tree",
+      "difficulty": 6,
```
[the remark this answers is not in the document]
Comment (amscotti) [reply config.json]: @Stargator Updated to `2`.

**Comment by Stargator** [comment]
@amscotti This needs to be rebased.

**Comment by amscotti** [comment]
@Stargator Rebased with master, also as a side note the Unit tests are starting to timeout on travis-ci.

**Comment by Stargator** [comment]
@amscotti Thanks for the update. I think we have an issue related to that. Basically, just testing what has been changed.

**Pull request merged**